### PR TITLE
Include xs1.h in rtos_swmem.c and update documentation

### PR DESCRIPTION
**Pull request by lucianomartin**
Fix tested with application https://github.com/ACascarino/tracealyzer_burn_trial/tree/broken.

Closes https://xmosjira.atlassian.net/browse/AP-94
Closes https://xmosjira.atlassian.net/browse/AP-224

### CODE REVIEW

**Review by ACascarino — APPROVED** [review]
Fixes bug observed :)

File: tools/ci/build_rtos_tests.sh [hunk 1]
```diff
@@ -41,6 +41,6 @@ for ((i = 0; i < ${#applications[@]}; i += 1)); do
 
     (cd ${path}; rm -rf build_${board})
     (cd ${path}; mkdir -p build_${board})
-    (cd ${path}/build_${board}; log_errors cmake ../ -G "$CI_CMAKE_GENERATOR" -DCMAKE_TOOLCHAIN_FILE=${toolchain_file} -DBOARD=${board} -DFRAMEWORK_RTOS_TESTS=ON; log_errors $CI_BUILD_TOOL ${make_target} $CI_BUILD_TOOL_ARGS)
+    (cd ${path}/build_${board}; log_errors cmake ../ -G "$CI_CMAKE_GENERATOR" --toolchain=${toolchain_file} -DBOARD=${board} -DFRAMEWORK_RTOS_TESTS=ON; log_errors $CI_BUILD_TOOL ${make_target} $CI_BUILD_TOOL_ARGS)
```
Comment (ACascarino) [inline tools/ci/build_rtos_tests.sh:44]: This sets minimum CMake version at 3.21, is this correct?

**Review by ACascarino — APPROVED** [review]
Happy for these changes to happen but as discussed a separate issue should be raised to cover the documentation changes.

CMake question nonwithstanding, happy to approve

File: tools/ci/build_rtos_tests.sh (hunk 1 above)
Comment (lucianomartin) [reply tools/ci/build_rtos_tests.sh:44]: That is the default minimum version in the CMakeLists.txt file. Is there any issue with that?

**Pull request merged**